compiler.ts

```diff
@@ -190,7 +208,42 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
 function codeGenValue(val: Value<Type>, env: GlobalEnv): Array<string> {
   switch (val.tag) {
     case "num":
-      return ["(i32.const " + val.value + ")"];
```

Comment:
Would it be possible to implement this all in `lower`, using `alloc` and `store` and `wasmint`? What are the tradeoffs with doing that?

Reply:
The current design allows constant folding in optimization. If bignums were lowered into multiple `wasmint`s and `store`s, it would create extra cost to reconstruct the constant value. I believe it would be a lot harder to maintain.

The benefit of implementing this in `lower` is that binop functions can be optimized. This can be approximated by writing good quality binop functions in `compiler.ts`.

Other than that, we are discussing with the memory management group to support our calls to `memory.wat`.